options/posix/generic/netdb-stubs.cpp

```diff
@@ -53,52 +55,63 @@ int getaddrinfo(const char *__restrict node, const char *__restrict service,
 	if (!node && !service)
 		return EAI_NONAME;
 
-	if (service) {
-		__ensure(!"Not implemented");
-		__builtin_unreachable();
-	}
-
 	int socktype = 0, protocol = 0;
 	if (hints) {
 		socktype = hints->ai_socktype;
 		protocol = hints->ai_protocol;
 	}
 
-	frg::vector<struct mlibc::dns_addr_buf, MemoryAllocator> buf(getAllocator());
-	int ret = mlibc::lookup_name_dns(buf, node);
-	if (ret < 0)
-		return ret;
-
-	auto out = (struct mlibc::ai_buf *) calloc(ret, sizeof(struct addrinfo));
-
-	auto canon_len = strlen(node);
-	auto canon = (char *) malloc(canon_len);
-	strncpy(canon, node, canon_len);
+	struct mlibc::service_buf serv_buf[SERV_MAX] = {};
+	int serv_count = mlibc::lookup_serv(serv_buf, service, protocol, socktype);
```

Comment:
It's guaranteed that this function only ever writes SERV_MAX = 2 entries? (If not, we could use a frg::small_vector or so.)

Reply:
yeah it is